tests/js/client/shell/transaction/shell-transaction.js

```diff
@@ -69,6 +70,30 @@ var sortedKeys = function (col) {
   return keys;
 };
 
+const getDBServers = function (db) {
+  let requestResult = db._connection.GET("/_admin/cluster/health");
+  arangosh.checkRequestResult(requestResult);
+  let dbServers = {};
+  for (const [serverId, serverInfo] of Object.entries(requestResult.Health)) {
+    if (serverInfo.Role === "DBServer") {
+      dbServers[serverId] = serverInfo.Endpoint.replace(/^tcp:/, 'http:').replace(/^ssl:/, 'https:');
+    }
+  }
+  return dbServers;
+};
+
+const getLocalValue = function (endpoint, db, col, key) {
+  let res = request.get({
+    url: `${endpoint}/_db/${db}/_api/document/${col}/${key}`,
+    headers: {
+      "X-Arango-Allow-Dirty-Read": true
+    },
+  });
+  arangosh.checkRequestResult(res);
+  return res.json;
+};
```

Comment:
I am pretty sure, that for replication 2 we have to add some index value, that we put into `waitForApplied`. Similar to what we did for the prototype state. This implies that the transaction operations somehow return this index.